x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/bundled_prebuilt_rules_package/trial_license_complete_tier/install_latest_bundled_prebuilt_rules.ts

```diff
@@ -31,7 +31,7 @@ export default ({ getService }: FtrProviderContext): void => {
   /* from a package that was bundled with Kibana */
   //
   // FLAKY: https://github.com/elastic/kibana/issues/180087
-  describe.skip('@ess @serverless @skipInQA install_bundled_prebuilt_rules', () => {
+  describe.skip('@ess @serverless @skipInServerless install_bundled_prebuilt_rules', () => {
```

Comment:
@MadameSheema We don't want all the tests for prebuilt rules to be skipped in all quality gates. We need them to run on CI, where, as far as I'm aware, they are not currently skipped and have been passing.

This also applies to some of the other rule management tests that are being skipped in CI by this PR.

I would suggest to:

1. Either introduce an additional `@skipInServerlessMKI` tag and use it for these tests instead of `@skipInServerless`. This tag would skip a test in the periodic pipeline + 2nd quality gate + 3rd and 4th quality gates in the future, but keep the test running on CI in the Serverless mode.
2. Or to rename the `@serverless` tag to `@serverlessCI` and introduce:
  - an additional tag `@serverlessRelease` that would add a test to all the release pipelines: the 2nd quality gate + 3rd and 4th quality gates in the future
  - an additional tag `@serverlessPeriodic` that would add a test only to the periodic pipeline

I'd personally choose the 2nd option as being more explicit about where a test should run.

Sorry if I missed this in https://github.com/elastic/kibana/pull/179737 where Cypress tests for prebuilt rules and rule management have been skipped in Serverless CI. We'd need to run those Cypress tests in CI too, so I'd propose the same changes for Cypress tests.

cc @jpdjere @maximpn @xcrzx

Reply:
@maximpn @yctercero since you were the ones against having extra labels, can you please arrive to an agreement about how to proceed with it?

This is my proposal:

@serverless -> Executes tests on CI and periodic pipeline (MKI)  
@serverlessQA -> Executes tests on the second quality gate (QA environment)  @skipServerless-> Does not execute the test in any serverless environment 
@skipServerlessMKI -> Does not execute the test in any MKI environment (periodic pipeline and quality gate)  

If you don’t want a test to be executed as part of the second quality gate you don’t add the `@serverlessQA` tag to it.

Reply:
@MadameSheema do I get it correctly a test which should be skipped in MKI env but run everywhere else will have the following tags

```ts
describe('@ess @serverless @serverlessQA @skipServerlessMKI install_bundled_prebuilt_rules', () => {
```

Here we return to the situation when it equals to

```ts
describe('@ess @serverless install_bundled_prebuilt_rules', () => {
```

Here we have two options

- Have only `@ess`, `@skipInEss`, `@serverless`, `@serverlessQA` and `@skipInServerless` tags. When we need to run a test in another MKI end we would add another `@serverless<name>` tag. By default we just don't add tags so the test isn't enabled automatically for any MKI environment. There is an assumption MKI env is less stable than mocked Serverless env. It's impossible to disabled a test for only mocked Serverless env but still run it against some MKI env. But it doesn't look like an issue.
- Have only `@ess`, `@skipInEss`, `@serverless`, `@skipInServerlessMock` (what's the best name for local/CI Serverless instead of serverless mock?) and `@skipInServerlessQA` tags. In this case tests tagged by `@serverless` will run against everything (besides ESS) and adding a corresponding `@skipInServerless<name>` tag skips a test in a proper env.

The second approach looks better for automatic skipping script. When a test fails in some env or becomes flaky it could be skipped automatically.

I checked the PR's diff once more time and see that it's more convenient to use the second approach. Quite often `describe` section has permissive tags `@ess` and `@serverless` while individual tests have may be skipped via `@skipInServerless`. It will require manually skip all the tests in MKI env via `@skipInServerlessQA` but it should be simpler to maintain the tests.

Reply:
@maximpn 

> Have only @ess, @skipInEss, @serverless, @serverlessQA and @skipInServerless tags.

This set of tags doesn't allow to run a test on CI in ESS + on CI in Serverless, but not run it in the periodic pipeline + release gates.

At this point, we need to be able to run many of our tests ONLY on CI in ESS + on CI in Serverless, because they are broken in MKI environments. Fixing this requires time and effort we don't have at the moment.

> Have only @ess, @skipInEss, @serverless, @skipInServerlessMock (what's the best name for local/CI Serverless instead of serverless mock?) and @skipInServerlessQA tags

This set of tags also doesn't allow to run a test on CI in ESS + on CI in Serverless, but not run it in the periodic pipeline + release gates.

We don't need to be able to skip a test on CI in Serverless, but run it in MKI pipelines. We don't have such class of tests as of now.

Reply:
> @banderror @MadameSheema

Could you clarify the requirements to running tests in periodic pipeline? Do we need a separate way to skip tests for periodic pipeline?

In my understanding if a test is broken in MKI env it will be broken in periodic pipeline since it's an MKI env as well. Taking this into account a test shouldn't have @serverlessQAtag (option 1) or have @skipInServerlessQA tag (option 2). It won't run against any MKI env until it's fixed, tested manually and tag manipulation is performed to run the test against MKI envs.

As there are differences between the serverless environment we use on CI and a real one but at the same time, we don't want to execute all the tests in the quality gate, is a way for us to determine the health of our application in a real serverless project environment.

A test with just `@serverless` is going to be executed in CI and periodic pipeline. If you add to the test `@serverlessQA` that test will be executed in the second quality gate as part of the release process. If a test is broken on MKI is going to fail in both on periodic and quality gate, so you can skip it with `@serverlessMKI` but it will still be executed on CI as part of the PRs so I strongly believe that right now what it makes sense for us to have as labels is:

```
@ess: Test is executed on CI ess environment
@skipEss: Test is excluded fom the CI ess environment execution
@serverless: Test is executed on CI serverless and periodic pipeline (MKI environment)
@serverlessQA: Test is executed as part of the release process on the second quality gate which is the QA environment (MKI environment)
@skipServerlessMKI: Test is excluded from being executed in any MKI environment, meaning, is not executed as part of the periodic pipeline and is not executed as part of the release process
@skipServerless: Tests is excluded from being executed as part from the CI and is not executed in any MKI environment
```

Reply:
@MadameSheema Your proposal works for me 👍

Reply:
Above proposal looks great to me @MadameSheema

Reply:
@MadameSheema

Me and @banderror discussed the approach on a zoom call. Tags you suggest in the last comment look good to me. I have some comments though

ESS tags are pretty clear. We need to run tests against ESS env and skip broken tests.
- `@ess`
- `@skipEss`

Serverless in general is pretty clear as well. We need to run test against Serverless env (abstractly it doesn't matter which one) and skip broken tests
- `@serverless`
- `@skipServerless`

If we don't need to run tests against some env an appropriate tag `@ess` or `@serverless` isn't added.

The most interesting part start from the fact that Serverless envs are different. Not all tests work in Serverless MKI env currently. Taking into account this fact we need an ability to skip such tests so we can use the following tag
- `@skipServerlessMKI`

And the last part is that we need some test to block Serverless release when critical functionality doesn't work. In this case something explicit like `@blockServerlessRelease` or `@serverlessRelease` would work the best
- `@serverlessRelease`

I have a question about your suggestion. Why do you want to use explicit `@serverlessQA`. Is it expected we are gonna run different test sets/subsets in `QA`, `Staging` etc Serverless MKI envs? It looks like if a test work in one Serverless MKI env it should work in the other one as well. Is there some difference between `QA` and `Staging`?

cc @rylnd

Reply:
Maybe at this point it's best to jump on a Zoom. I'm a bit confused as to what's being asked here anymore. 

@MadameSheema worked on this PR after we chatted and decided it'd be great for our API and cypress tags to match so this PR brought them into parity. We had discussed and decided on the tags during the first PR that changed the Cypress tags. It would be great to finalize the discussion so we don't keep rehashing the discussion in each PR.